azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkits/appservice/model/Runtime.java

```diff
@@ -34,6 +37,91 @@
     public static final Runtime LINUX_JAVA11_TOMCAT85 = new Runtime(OperatingSystem.LINUX, WebContainer.TOMCAT_85, JavaVersion.JAVA_11);
     public static final Runtime DOCKER = new Runtime(OperatingSystem.DOCKER, null, null);
 
+    public static Runtime getRuntime(OperatingSystem operatingSystem, WebContainer webContainer, JavaVersion javaVersion) {
+        return values().stream()
+                .filter(runtime -> runtime.operatingSystem == operatingSystem)
+                .filter(runtime -> runtime.webContainer == webContainer)
+                .filter(runtime -> runtime.javaVersion == javaVersion)
+                .findFirst().orElse(new Runtime(operatingSystem, webContainer, javaVersion));
+    }
+
+    public static Runtime getRuntimeFromWebApp(WebAppBase webAppBase) {
+        if (StringUtils.startsWithIgnoreCase(webAppBase.linuxFxVersion(), "docker")) {
+            return DOCKER;
+        }
+        return webAppBase.operatingSystem() == com.azure.resourcemanager.appservice.models.OperatingSystem.WINDOWS ?
+                getRuntimeFromWindowsWebApp(webAppBase) : getRuntimeFromLinuxWebApp(webAppBase);
+    }
+
+    public static Runtime getRuntimeFromLinuxWebApp(WebAppBase webAppBase) {
+        if (StringUtils.isEmpty(webAppBase.linuxFxVersion())) {
+            return null;
+        }
+        final String linuxFxVersion = webAppBase.linuxFxVersion().replace("|", " ");
+        return getRuntimeFromLinuxFxVersion(linuxFxVersion);
+    }
+
+    public static Runtime getRuntimeFromLinuxFxVersion(String linuxFxVersion) {
+        final JavaVersion javaVersion = StringUtils.containsIgnoreCase(linuxFxVersion, "java11") ? JavaVersion.JAVA_11 : JavaVersion.JAVA_8;
+        final WebContainer webContainer = WebContainer.values().stream()
+                .filter(container -> StringUtils.containsIgnoreCase(linuxFxVersion, container.getValue()))
```

Comment:
too many codes like this:
xx.values().stream()
                .filter(container -> StringUtils.containsIgnoreCase(linuxFxVersion, container.getValue()))
                .findFirst().orElse(null);

Reply:
We need to convert Runtime to multi sdk model or get runtime from sdk, so we may list all values and get correspond value, will check whether we could extract duplicate codes later